### ISSUE TITLE
Suggestion: NavMesh object / Obstacle object

### ISSUE THREAD

**Issue by rafcolson** [GoranM/bdx#788]
Hi,

In the future I'd like to add support for Blender Navigation Meshes / Obstacles. It will be a challenge reverse engineering this (I'm learning C++ to be able to analyze the Ketsji code). I've done a little research already and I think it should be possible. So my question:

If it will work, is it okay to add this to master?

**Comment by SolarLune** [comment]
Yeah, sure, I'm OK with it. Some method to recalculate the navmesh would be nice, in case you edit it or merge two navmeshes together.

**Comment by GoranM** [comment]
I think the feature would be useful, but I don't think we can really promise that something will be added to master, because there's more than a few factors that go into that decision, a lot of which can only be known when there's actually a PR to review.

If it's something that you find interesting, I think you should work on it - It might fit nicely into BDX, but even if it doesn't, you'll probably learn enough new things to make the effort worthwhile.

**Comment by SolarLune** [comment]
That's true, navmeshes do seem pretty complex - I don't know much about them, though. It might be good to leverage the existing vertex manipulation functions, or even add polygon manipulation functions to the Mesh class to help ease this? Just spitballin' here, but if you know where each triangle is and where its boundaries are for an ordinary Mesh, then it seems like you could more easily get a visualization and representation of the outer edges of a NavMesh.

There are readups on NavMeshes that should be of use, anyway.

**Comment by rafcolson** [comment]
I've found a lot already, but if you find anything that you think might be useful to look into, you're more than welcome to post some links here. It is pretty complex, so it will take a while for me to do it. And yes, it will make the effort worthwhile, even if for any reason it wouldn't get to master. But I'm happy that you guys like the idea. I'm leaving this open.
